### PR TITLE
MAKE-1292: benchmark bucket implementations

### PR DESCRIPTION
**Pull request by Kimchelly**
JIRA: https://jira.mongodb.org/browse/EVG-8980

* Vendor poplar.
* Write benchmarks for push/pull. They aren't super well tested because can't be easily run locally and we also can't use the benchmark results yet (but there's a ticket to support uploading poplar benchmark results from evergreen tasks to cedar eventually).

### CODE REVIEW

File: benchmarks/harness.go [hunk 1]
```diff
@@ -0,0 +1,364 @@
...
+	"github.com/evergreen-ci/pail/testutil"
+	"github.com/evergreen-ci/poplar"
+	"github.com/evergreen-ci/utility"
+	"github.com/mongodb/grip"
+	"github.com/pkg/errors"
+)
+
+func buildDir() string {
+	_, file, _, _ := runtime.Caller(0)
+	return filepath.Join(filepath.Dir(filepath.Dir(file)), "build")
```
Comment (julianedwards) [inline benchmarks/harness.go:22]: Shouldn't this be one more level up? Or is caller root?

Comment (Kimchelly) [reply benchmarks/harness.go:22]: `filepath.Dir` gives you the directory of the current file (i.e. `benchmarks/harness.go/..`)

File: makefile [hunk 2]
```diff
@@ -115,6 +123,13 @@ vendor-clean:
 	rm -rf vendor/go.mongodb.org/mongo-driver/vendor/github.com/montanaflynn
 	rm -rf vendor/github.com/mongodb/grip/vendor/github.com/montanaflynn
 	rm -rf vendor/github.com/mongodb/grip/vendor/github.com/pkg/errors
+	rm -rf vendor/github.com/evergreen-ci/poplar/vendor/github.com/mongodb/grip
+	rm -rf vendor/github.com/evergreen-ci/poplar/vendor/github.com/pkg/errors
+	rm -rf vendor/github.com/evergreen-ci/poplar/vendor/github.com/stretchr/testify
+	rm -rf vendor/github.com/evergreen-ci/poplar/vendor/go.mongodb.org/mongo-driver
+	rm -rf vendor/github.com/evergreen-ci/poplar/vendor/github.com/mongodb/ftdc/vendor/go.mongodb.org/mongo-driver/
+	# Note: we have a circular dependency problem between pail and poplar
```
[the remark this answers is not in the document]
Comment (Kimchelly) [reply makefile:131]: Idk ¯\_(ツ)_/¯

File: benchmarks/harness.go (hunk 1 above)
Comment (julianedwards) [reply benchmarks/harness.go:22]: Missed the double filepath.Dir! Thanks

**Pull request merged**